src/lib/Helpers/ParsingHelpers.cs

```diff
@@ -136,6 +138,41 @@ internal static SortedSet<string> GetOrderedHashSetOfString(JsonElement v)
             yield return keyValue;
         }
     }
+
+    internal static async Task<ReadResult> ParseOpenApiAsync(string openApiFileUrl, bool inlineExternal, CancellationToken cancellationToken)
+    {
+        Stream stream = await GetStreamAsync(openApiFileUrl, cancellationToken);
+        ReadResult result = await new OpenApiStreamReader(new OpenApiReaderSettings
+        {
+            LoadExternalRefs = inlineExternal,
+            BaseUrl = new Uri(openApiFileUrl)
+        }
+        ).ReadAsync(stream, cancellationToken);
+
+        return result;
+    }
+
+    private static async Task<Stream> GetStreamAsync(string input, CancellationToken cancellationToken = default)
+    {
+        if (!input.StartsWith("http"))
```

Comment:
I opted to support both schemes since we don't require HTTPS in the API manifest spec - https://www.ietf.org/archive/id/draft-miller-api-manifest-01.html#name-api-dependency-object. @darrelmiller, is this something we should call out in the API manifest spec?